### PR TITLE
Enable XFS testing on NVMe devices

### PR DESCRIPTION
**Pull request by hreinecke**
In order to do regression testing for my large-block work this patchset enables XFS tests to run on NVMe devices.

### CODE REVIEW

**Comment by hreinecke** [comment]
And, as you might have noticed, it's based on the 'su -l' patches, so it probably should be rebased once they are merged.

**Comment by ddiss** [comment]
I'll queue up the obvious changes and push them soon. I'm still a bit hesitant on the idempotent vm_autorun and modprobe changes so will leave those on the backburner.

File: cut/fstests_xfs.sh [hunk 1]
```diff
@@ -45,7 +45,7 @@ req_inst+=("/usr/include/xfs/*.h")
 		   ${FSTESTS_SRC}/src/log-writes/* \
 		   ${FSTESTS_SRC}/src/aio-dio-regress/*" \
 	--include "$FSTESTS_SRC" "$FSTESTS_SRC" \
-	--add-drivers "zram lzo lzo-rle dm-snapshot dm-flakey xfs \
+	--add-drivers "zram nvme lzo lzo-rle dm-snapshot dm-flakey xfs \
```
Comment (ddiss) [inline cut/fstests_xfs.sh:48]: how are you passing through the nvme devices to qemu - any chance you could add an example to `rapido.conf.example` in future?

Comment (hreinecke) [reply cut/fstests_xfs.sh:48]: -drive id=test,file=/tmp/fstests-test.qcow2,if=none,cache=none -device virtio-blk-pci,drive=test,bus=pci.0,serial=TEST_DEV

Comment (hreinecke) [reply cut/fstests_xfs.sh:48]: Easy, really :-)

File: autorun/lib/fstests.sh [hunk 2]
```diff
@@ -77,6 +79,9 @@ _fstests_devs_pool_provision() {
 _fstests_users_groups_provision() {
 	local ug xid="2000"
 
+	echo "daemon:x:2:2:Daemon:/:/sbin/nologin" \
+	     >> /etc/passwd
+	echo "daemon:x:2:" >> /etc/group
 	for ug in fsgqa fsgqa2 123456-fsgqa; do
```
Comment (ddiss) [inline autorun/lib/fstests.sh:85]: if the uid/gid=2 value isn't needed then you could just add `daemon` to the loop alongside `fsgqa`, etc.

**Comment by ddiss** [comment]
> I'll queue up the obvious changes and push them soon. I'm still a bit hesitant on the idempotent vm_autorun and modprobe changes so will leave those on the backburner.

most of these are queued up in my `hare-nvme-queued` branch for now

**Comment by ddiss** [comment]
@hreinecke I've submitted an alternative to your idempotence change via https://github.com/rapido-linux/rapido/pull/239 . Would be great if you could take a quick look.

> > I'll queue up the obvious changes and push them soon. I'm still a bit hesitant on the idempotent vm_autorun and modprobe changes so will leave those on the backburner.
> 
> most of these are queued up in my `hare-nvme-queued` branch for now

These all went into master.

File: vm_autorun.env [hunk 3]
```diff
@@ -274,6 +278,7 @@ _vm_ar_ip_addrs_nomask() {
 _vm_ar_load_kmods() {
 	local -a kmods
 
+	which modprobe &> /dev/null || return
```
Comment (ddiss) [inline vm_autorun.env:281]: Given that this was only needed for subsequent `su -` login shells, I take it we can drop this change with `vm_autorun.sh` moved out of the profile path? I.e. https://github.com/rapido-linux/rapido/pull/239

Comment (ddiss) [reply vm_autorun.env:281]: I've merged https://github.com/rapido-linux/rapido/pull/239 as an alternative to the idempotence change; other commits went in. I think this PR can probably be closed.

**Comment by ddiss** [comment]
> I've merged https://github.com/rapido-linux/rapido/pull/239 as an alternative to the idempotence change; other commits went in. I think this PR can probably be closed.

Closing...

**Pull request closed without merging**